pkg/appfile/parser.go

```diff
@@ -112,6 +112,12 @@ func (p *Parser) GenerateAppFileFromApp(ctx context.Context, app *v1beta1.Applic
 	ns := app.Namespace
 	appName := app.Name
 
+	for idx := range app.Spec.Policies {
+		if app.Spec.Policies[idx].Name == "" {
+			app.Spec.Policies[idx].Name = fmt.Sprintf("%s:auto-gen:%d", app.Spec.Policies[idx].Type, idx)
```

Comment:
Does K8s accept the name contains the `:`?

Reply:
This is a internal name, not stored in etcd. FYI, it supports ":".

Reply:
It can be used in `context.name`?

Reply:
No. There is no need for `context.name`. `context.name` currently refers to either `workflowstep` or `component`.